### ISSUE TITLE
very  slow  work  on android 11  amlogic s905w2

### ISSUE THREAD

**Issue by leksmax** [Genymobile/scrcpy#3936]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: lsb_release -a
No LSB modules are available.
Distributor ID:	Linuxmint
Description:	Linux Mint 21
Release:	21
Codename:	vanessascrcpy --version
#######################
- scrcpy version: 
scrcpy 2.0 <https://github.com/Genymobile/scrcpy>
Dependencies (compiled / linked):
 - SDL: 2.0.20 / 2.0.20
 - libavcodec: 58.134.100 / 58.134.100
 - libavformat: 58.76.100 / 58.76.100
 - libavutil: 56.70.100 / 56.70.100
 - libavdevice: 58.13.100 / 58.13.100
 - libusb: - / 1.0.25
#########################
 - installation method: manual build
 - device model:XS97( amlogic s905w2)
 - Android version:  android 11
#########################

very slow scrcpy. I connect via ip address.Adb latest version .

**Comment by rom1v** [comment]
Is it slow over USB?

**Comment by leksmax** [comment]
cant connect via  usb now.
in  logcat  many mesages  like  this
04-21 14:00:53.020   849   849 I ThermalService: CPU temperatures: [56.500004]
04-21 14:00:53.036  3000  3023 D CCodecBuffers: [c2.android.opus.encoder#498:Output[N]] popFromStashAndRegister: output format changed to AMessage(what = 0x00000000) = {
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   int32_t bitrate = 128000
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   int32_t channel-count = 2
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   int32_t flac-compression-level = 10
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   int32_t max-bitrate = 128000
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   string mime = "audio/opus"
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   int32_t sample-rate = 48000
04-21 14:00:53.036  3000  3023 D CCodecBuffers:   Buffer *csd-0 = 0xebb0fbd0
04-21 14:00:53.036  3000  3023 D CCodecBuffers: }

**Comment by rom1v** [comment]
If you record (`scrcpy --record=file.mp4`), is the recorded file "laggy"?

**Comment by leksmax** [comment]
yes, the video recording is also delayed. But on the device itself there is no delay when navigating (up-down, left-right)

**Comment by rom1v** [comment]
> the video recording is also delayed

How do you observed that it is "delayed" when you play the recorded file?

Could you share such a recorded file (for a few seconds)?

If it is laggy, then the problem is on the device video capture.

Does it work better with a lower resolution? (for example `scrcpy -m1024`)?

**Comment by leksmax** [comment]
with lower resolution the speed is slightly faster
https://drive.google.com/file/d/1JHeGEc1tGNEIGU7P7ygXPlKhfrshyokc/view?usp=share_link

**Comment by rom1v** [comment]
OK so you have a low capture framerate. It's a problem on the device encoder.

Try with another [encoder](https://github.com/Genymobile/scrcpy/blob/master/doc/video.md#codec) if several are available (`scrcpy --list-encoders`).

**Comment by leksmax** [comment]
scrcpy  --list-encoders
scrcpy 2.0 <https://github.com/Genymobile/scrcpy>
/usr/local/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 177.6 MB/s (52867 bytes in 0.000s)
[server] INFO: List of video encoders:
    --video-codec=h264 --video-encoder='c2.android.avc.encoder'
    --video-codec=h264 --video-encoder='OMX.google.h264.encoder'
[server] INFO: List of audio encoders:
    --audio-codec=opus --audio-encoder='c2.android.opus.encoder'
    --audio-codec=aac --audio-encoder='c2.android.aac.encoder'
    --audio-codec=aac --audio-encoder='OMX.google.aac.encoder'

with any codec it is also slowly displayed on the computer screen and fps is about 10

**Comment by leksmax** [comment]
Is there any solution to this encoder problem? do i need to patch scrcpy or add another encoder to android tv box?

**Comment by rom1v** [comment]
This is probably a hardware problem on the device itself (a tv box may not have an efficient encoder because this is not useful like when there is a camera).

**Comment by Montesuma80** [comment]
Set the HDMI Resolution to 1080P or 720P then it works fast
